### ISSUE TITLE
False positives when using explicit_type_interface with allowRedundancy

### ISSUE THREAD

**Issue by Dschee** [realm/SwiftLint#2425]
Using the lastest SwiftLint version, when I use the rule `explicit_type_interface` with this configuration

```yaml
explicit_type_interface:
  allow_redundancy: true
```

most of the cases are handled correctly. But for some initializers Swift has some sugar syntax integrated and some initializers might throw. In both cases this option doesn't work correctly.

Specifically, the following three lines are shown as warnings but shouldn't:

```swift
let regex = try! NSRegularExpression(pattern: ".*")
let array = [String]()
let dict = [String: String]()
```
